### ISSUE TITLE
Temperature explode during OTF learning

### ISSUE THREAD

**Issue by hhlim12** [mir-group/flare#280]
Dear FLARE Developers,

Thank you for the FLARE,

I’d like to ask question about the calculation of Al adatom on Al(111) surface as in Fig.4d in the paper (Vandermause, J., Torrisi, S.B., Batzner, S. et al. On-the-fly active learning of interpretable Bayesian force fields for atomistic rare events. npj Comput Mater 6, 20 (2020). https://doi.org/10.1038/s41524-020-0283-z).

I have tried the calculation with the latest version of FLARE (as Feb, 25th 2021). However, my calculation showed the temperature explode and never stabilize. Also, the error keep increasing even the database is keep updated and the likelihood is high.
Do you have any suggestion where am I wrong and how to reproduce the result as in the paper ?
I attach my script and the output below. 

[Al_adatom.zip](https://github.com/mir-group/flare/files/6047428/Al_adatom.zip)

With best regards

**Comment by YuuuXie** [comment]
The temperature exploding issue looks similar to #273 and #274. Also it seems that the MAE of forces is very large, and especially GP predicts very large forces, which is kind of weird. But I think I might have seen this kind of issue before. I'll check it.

**Comment by jonpvandermause** [comment]
Hi @hhlim12,

Can you please try rerunning with `force_only=True` in the `ASE_OTF` constructor? The Quantum Espresso energies are extremely negative (around -20,000 eV) and can throw off the GP algebra if they're included in the training set without being recentered at a more reasonable value. (Note that in the paper we trained on forces only.)

Thanks,
Jon

**Comment by hhlim12** [comment]
Dear Yuu and Jon, 

I have recalculated using `force_only = True` and it’s work fine now. Thank you for your help and quick response. 
Previously I thought  FLARE  by default uses force  for training since there is no `force_only` parameter in the of ASE module (https://flare.readthedocs.io/en/latest/flare/ase/otf.html). 


Regarding the large negative energy, is it okay if I just shift the energy manually? and to what reasonable value should I shift it?. In the case of transition metals, most of the pseudopotentials give very large negative energy so I wonder whether such energies can be used or not.

Thank you very much.
I look forward to do research using FLARE.


With best regards,

**Comment by mkrompiec** [comment]
Can you please try rerunning with force_only=True in the ASE_OTF
> constructor? The Quantum Espresso energies are extremely negative (around
> -20,000 eV) and can throw off the GP algebra if they're included in the
> training set without being recentered at a more reasonable value. (Note
> that in the paper we trained on forces only.)
>
How about “recentering” them automatically? By subtracting energies of
isolated atoms (which would have to be provided by the user) from the
training energies?

>

**Comment by jonpvandermause** [comment]
@hhlim12,

Glad to hear it's working. `ASE_OTF` is a child class of `OTF` and can take any of the `OTF` keyword arguments as input, so you can find the documentation for the `force_only` keyword [here](https://flare.readthedocs.io/en/latest/flare/otf.html).

As @mkrompiec mentioned, it's best to train on the _cohesive energy_ rather than the raw energy, which requires you to subtract off the energy of a corresponding system of isolated atoms. This ensures that the cohesive energy of the model agrees with DFT, since the GP always assigns zero energy to isolated atoms. You can compute these isolated atom energies by running DFT on a single atom in a large box (usually 10 A or so is sufficient).

@mkrompiec, that's a good idea to do this automatically. I think I'll add an optional `single_atom_energies` attribute to the `GaussianProcess` class, and use this to shift the energy labels inside the `update_db` method.

Best,
Jon

**Comment by hhlim12** [comment]
@jonpvandermause 

Thank you for the answer. I'll try your suggestion. 
But anyway, so far training with force only already gives reasonable results for my system.
I will close this issue.

With best regards
